### PR TITLE
Handle tile data in workflows

### PR DESCRIPTION
**Pull request by chrabyrd**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Issues Solved

#8293

### CODE REVIEW

**Review by robgaston — APPROVED** [review]
These changes look good to me with the understanding that this is an incremental step towards improving workflows that doesn't break backwards compatibility.

**Comment by robgaston** [comment]
@chiatt could you please review these changes as well since you know the domain a bit better here?

File: arches/app/views/api.py [hunk 1]
```diff
@@ -1457,6 +1457,31 @@ def post(self, request, tileid):
...
+        except ValueError as e:
+            del params["nodegroupid"]
+
+        try:
+            nodegroup = models.NodeGroup.objects.get(pk=params["nodegroupid"])
+            permitted_nodegroups = [nodegroup.pk for nodegroup in get_nodegroups_by_perm(user, perms)]
+        except Exception as e:
+            return JSONResponse(str(e), status=404)
+
+        if not nodegroup or not nodegroup.pk in permitted_nodegroups:
```
Comment (chiatt) [inline arches/app/views/api.py:1479]: Isn't the result the same if it's written like this? I find the syntax clearer.
```not nodegroup or nodegroup.pk not in permitted_nodegroups```

**Review by chiatt — CHANGES_REQUESTED** [review]
This looks great. There are just a few minor Stickler errors. Could you resolve them real quick (unless there is a reason not to that I'm overlooking)?

File: arches/app/views/api.py (hunk 1 above)
Comment (chrabyrd) [reply arches/app/views/api.py:1479]: 👍

**Review by chiatt — APPROVED** [review]

**Pull request merged**